src/ts/components/DashSocketIO.tsx

```diff
@@ -1,64 +1,95 @@
+  options?: Partial<SocketOptions>;
   /** Name of Socket.IO events to listen to */
-  eventNames?: string[]
+  eventNames?: string[];
   /** Whether the client is connected to the websocket - READONLY */
-  connected?: boolean
+  connected?: boolean;
   /** The socket ID */
-  socketId?: string
+  socketId?: string;
 } & DashComponentProps;
+
+const defaultOptions: SocketOptions = {
+  path: '/socket.io',
+  forceNew: false,
+  autoConnect: true
+};
+
 /**
  * Socket.IO Dash component.
  */
 const DashSocketIO = (props: Props) => {
-    const { url, setProps, eventNames } = props;
-
-    const socket = React.useMemo(() => io(
-      url || undefined, {auth: {pathname: window.location.pathname}}
-    ), [url])
+  const { url, options = {}, setProps, eventNames } = props;
 
-    React.useEffect(() => {
-      const onConnect = () => {
-        setProps({connected: true, socketId: socket.id});
+  const socket = React.useMemo(() => io(
+    url || undefined,
+    {
+      ...defaultOptions,
+      ...options,
+      auth: {
+        ...options.auth,
+        pathname: window.location.pathname
       }
+    }
+  ), [url, options]);
 
-      const onDisconnect = () => {
-        setProps({connected: false, socketId: null});
-      }
+  React.useEffect(() => {
+    const onConnect = () => {
+      setProps({ connected: true, socketId: socket.id });
+    }
 
-      socket.on('connect', onConnect);
-      socket.on('disconnect', onDisconnect);
+    const onDisconnect = () => {
+      setProps({ connected: false, socketId: null });
+    }
 
-      return () => {
-        socket.off('connect', onConnect);
-        socket.off('disconnect', onDisconnect);
-        socket.disconnect();
-      };
-    }, [socket]);
+    socket.on('connect', onConnect);
+    socket.on('disconnect', onDisconnect);
 
-    React.useEffect(() => {
-      if (!eventNames || eventNames.length === 0) {
-        return;
-      }
-      function onEvent(eventName: string) {
-        return (data: any) => {
-          setProps({[`data-${eventName}`]: data});
-        }
+    return () => {
+      socket.off('connect', onConnect);
+      socket.off('disconnect', onDisconnect);
+      socket.disconnect();
+    };
+  }, [socket]);
+
+  React.useEffect(() => {
+    if (!eventNames || eventNames.length === 0) {
+      return;
+    }
+    function onEvent(eventName: string) {
+      return (data: any) => {
+        setProps({ [`data-${eventName}`]: data });
       }
-      eventNames.forEach((eventName) => socket.on(eventName, onEvent(eventName)));
-      return () => {
-        eventNames.forEach((eventName) => socket.off(eventName, onEvent(eventName)));
-      };
-    }, [socket, eventNames])
+    }
+    eventNames.forEach((eventName) => socket.on(eventName, onEvent(eventName)));
+    return () => {
+      eventNames.forEach((eventName) => socket.off(eventName, onEvent(eventName)));
+    };
+  }, [socket, eventNames])
 
-    return null
+  return null;
 }
 
-DashSocketIO.defaultProps = {};
```

Comment:
Any reason this was deleted?

Reply:
Not really. I defined default values with 
`const defaultOptions: SocketOptions = {
  path: '/socket.io',
  forceNew: false,
  autoConnect: true
};`

That's why I thought it might be not needed.

Reply:
I think Dash might be expecting this to exist, at least it was part of the template, I suggest rolling this back.